src/test/suite/port-allocation.test.ts

```diff
@@ -0,0 +1,105 @@
+import * as assert from "assert";
+import { TestPortManager } from "../helpers/test-port-manager";
+
+suite("Dynamic Port Allocation Tests", () => {
+  teardown(() => {
+    // Clean up after each test
+    TestPortManager.releasePort();
+  });
+
+  test("allocates available port in test range", async function () {
+    this.timeout(10000);
+
+    const port = await TestPortManager.allocateTestPort();
+    
+    // Verify port is in expected range
+    assert.ok(port >= 9043, `Port ${port} should be >= 9043`);
+    assert.ok(port <= 9999, `Port ${port} should be <= 9999`);
+    
+    // Verify port is not the production port
+    assert.notStrictEqual(port, 9042, "Test port should not be the production port 9042");
+    
+    console.log(`✓ Allocated test port: ${port}`);
+  });
+
+  test("detects production port usage", async function () {
+    this.timeout(5000);
+
+    const productionInUse = await TestPortManager.isProductionPortInUse();
+    console.log(`Production port (9042) in use: ${productionInUse}`);
+    
+    // This test just verifies the detection works - both true/false are valid
+    assert.ok(typeof productionInUse === "boolean", "Should return boolean");
+  });
+
+  test("provides correct test server URL", async function () {
+    this.timeout(10000);
+
+    const port = await TestPortManager.allocateTestPort();
+    const serverUrl = TestPortManager.getTestServerUrl();
+    
+    const expectedUrl = `http://localhost:${port}`;
+    assert.strictEqual(serverUrl, expectedUrl, `Server URL should be ${expectedUrl}`);
+    
+    console.log(`✓ Test server URL: ${serverUrl}`);
+  });
+
+  test("handles port allocation consistently", async function () {
+    this.timeout(10000);
+
+    const port1 = await TestPortManager.allocateTestPort();
+    const port2 = await TestPortManager.allocateTestPort();
+    
+    // Should return the same port when called multiple times
+    assert.strictEqual(port1, port2, "Should return same port on subsequent calls");
+    
+    console.log(`✓ Consistent port allocation: ${port1}`);
+  });
+
+  test("provides diagnostic information", async function () {
+    this.timeout(10000);
+
+    const port = await TestPortManager.allocateTestPort();
+    const diagnostics = await TestPortManager.getDiagnostics();
+    
+    assert.strictEqual(diagnostics.allocatedPort, port, "Diagnostics should show allocated port");
+    assert.ok(typeof diagnostics.allocatedPortAvailable === "boolean", "Should report port availability");
+    assert.ok(typeof diagnostics.productionPortInUse === "boolean", "Should report production port status");
```

Comment:
Because they will never use a production port, nothing in the tests need to know if it is in use.
```suggestion
```